solutions/src/router.js

```diff
@@ -0,0 +1,85 @@
+const getData = require('./queries/getData.js');
+const postData = require('./queries/postData.js');
+const queryString = require('querystring');
+
+const router = (request, response) => {
+    const endpoint = request.url.split('/')[1];
+
+    if (endpoint === '') {
+        fs.readFile(__dirname + "/../public/index.html", function(error, file) {
+            if (error) {
+                response.writeHead(500, 'Content-Type:text/html');
+                response.end('<h1>Sorry, there was a problem loading the homepage</h1>');
+                console.log(error);
+            } else {
+                response.writeHead(200, {
+                    "Content-Type": "text/html"
+                });
+                response.end(file);
+            }
+        });
+    } else if (endpoint === "users") {
+        // TASK 1: replace the 3 lines below below with your own function that gets data from your database
+        getData((err, res) => {
+            if (err) {
+                response.writeHead(500, 'Content-Type:text/html');
+                response.end('<h1>Sorry, there was a problem getting the users</h1>');
+                console.log(error);
+            } else {
+                let output = JSON.stringify(res);
+                response.writeHead(200, {
+                    'content-type': 'application/json'
+                });
+                response.end(output);
+            }
+        });
+    } else if (endpoint === "create-user") {
+        let data = '';
+        request.on('data', function(chunk) {
+            data += chunk;
+        });
+        request.on('end', () => {
+            const name = queryString.parse(data).name;
+            const location = queryString.parse(data).location;
+            postData(name, location, (err, res) => {
+                if (err) {
+                    response.writeHead(500, 'Content-Type:text/html');
+                    response.end('<h1>Sorry, there was a problem adding that user</h1>');
+                    console.log(err)
+                }
+            });
+            response.writeHead(200, {
+                "Content-Type": "text/html"
+            });
+            fs.readFile(__dirname + "/../public/index.html", function(error, file) {
+                if (error) {
+                    console.log(error);
+                    return;
+                } else {
+                    response.end(file);
+                }
+            });
+        });
+    } else {
+        const fileName = request.url;
+        const fileType = request.url.split(".")[1];
+        fs.readFile(__dirname + "/../public" + fileName, function(error, file) {
+            if (error) {
+                response.writeHead(500, 'Content-Type:text/html');
+                response.end('<h1>Sorry, there was a problem loading this page</h1>');
+                console.log(error);
+            } else {
+                response.writeHead(200, {
+                    "Content-Type": "text/" + fileType
```

Comment:
Is this router used to serve javascript and favicons? If so this will produce the incorrect content-type headers (`application/javascript` and `image/x-icon` instead of `text/js` and `text/ico`, respectively). The [node-intro-workshop](https://github.com/foundersandcoders/Node-Intro-Workshop/blob/mentor-walkthrough/step-5/src/handlers.js#L18-L38) has an example solution for this kind of handler.